src/components/Dropdown/Dropdown.tsx

```diff
@@ -628,7 +628,11 @@ const Dropdown = ({
       // when searchable, only blur if the event is from the input
       setFocusTimeoutId(
         window.setTimeout(() => {
-          if (focusWithin && (!searchable || e.target.id === `${name}-search-input`)) {
+          if (
+            focusWithin &&
+            ((!searchable && e.target.id === `${name}-dropdown-button`) ||
+              (searchable && e.target.id === `${name}-search-input`))
```

Comment:
this is getting kind of big, could you make a variable above the if that encapsulate the logic here? Like
```ts
const blurredFromButton = (!searchable && e.target.id === `${name}-dropdown-button`);
etc
```

Reply:
Done 👍🏼